### PR TITLE
Add csv.markdown

### PR DESCRIPTION
**Pull request by siom79**
**Description:**
Documentation for pull request [17296](https://github.com/home-assistant/home-assistant/pull/17296).

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#17296

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

### CODE REVIEW

File: source/_components/csv.markdown [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+To use this component in your installation, add the following to your `configuration.yaml` file:
+
+## {% linkable_title Configuration %}
+
+```yaml
+# Example configuration.yaml entry
+csv:
+  data_dir: '/home/homeassistant/data_dir'
+  separator: ','
```
Comment (frenck) [inline source/_components/csv.markdown:25]: We try to keep all configuration samples minimal. Thus, no optional requirement in the default sample. This helps a user to get started quickly by copy-&-paste the sample without worrying about optional parameters which they most likely not need. If required, insert a full configuration sample later that covers special setups or alike.

File: source/_components/csv.markdown [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+title: "CSV History"
+description: "Instructions on how to store events in CSV files."
+date: 2018-10-09 21:13
+sidebar: true
+comments: false
+sharing: true
+footer: true
+logo: home-assistant.png
+ha_category: History
+ha_release: 0.80
```
Comment (frenck) [inline source/_components/csv.markdown:12]: `0.80` has already been cut off, so this should be `0.81`.

File: source/_components/csv.markdown [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  required: true
+  type: string
+separator:
+  description: The separator between the values inside the CSV file. Default value is ','.
+  required: false
+  type: string
+gzip:
+  description: If set to `True`, old files are compressed using gzip. Default value is `True`.
+  required: false
+  type: boolean
```
Comment (frenck) [inline source/_components/csv.markdown:52]: A boolean always should have a `default: ` key.

File: source/_components/csv.markdown [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+purge_keep_days:
+  description: Sets the number of old files to keep. Older files are removed. Default value is 10.
```
Comment (frenck) [inline source/_components/csv.markdown:54]: Default should be listed in a separate `default: ` key.

File: source/_components/csv.markdown [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      - domain_to_exclude
+```
+
+{% configuration %}
+data_dir:
+  description: An existing directory where the CSV files are stored.
+  required: true
+  type: string
+separator:
+  description: The separator between the values inside the CSV file. Default value is ','.
```
Comment (frenck) [inline source/_components/csv.markdown:46]: Default should be listed in a separate `default: ` key.

**Review by frenck — CHANGES_REQUESTED** [review]
@siom79 Thanks for the PR!
I've reviewed it and left some comments. Could you please take a look? 👍

**Comment by siom79** [comment]
Closed because the parent PR has been closed.

**Pull request closed without merging**